src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -23,17 +25,27 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
             await this.launchApp(folder, configuration);
         }
 
-        const result = await vscode.commands.executeCommand<{
-            url: string,
-            inspectUri: string,
-            debuggingPort: number,
-        }>('blazorwasm-companion.launchDebugProxy', folder, configuration);
+        let inspectUri = '{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}';
+        let url = 'https://localhost:5001';
+        try {
+            if (folder !== undefined) {
+                let folderPath = configuration.cwd ? configuration.cwd : fileURLToPath(folder.uri.toString());
+                folderPath = folderPath.replace('${workspaceFolder}', fileURLToPath(folder.uri.toString()));
+                const launchSettings = JSON.parse(readFileSync(join(folderPath, 'Properties', 'launchSettings.json'), 'utf8'));
+                if (launchSettings?.profiles && launchSettings?.profiles[Object.keys(launchSettings.profiles)[0]]?.inspectUri) {
+                    inspectUri = launchSettings.profiles[Object.keys(launchSettings.profiles)[0]].inspectUri;
+                    url = launchSettings.profiles[Object.keys(launchSettings.profiles)[0]].applicationUrl.split(';', 1)[0];
```

Comment:
Nice to see that we will be trying to get the URL from the launchSettings going forward 👍